pilot/pkg/model/context.go

```diff
@@ -578,8 +568,9 @@ func (node *Proxy) SetWorkloadLabels(env *Environment) error {
 		log.Errorf("failed to get service proxy labels: %v", err)
 		return err
 	}
-
-	node.WorkloadLabels = l
+	if len(l) > 0 {
+		node.Metadata.Labels = l[0]
```

Comment:
it seems to be that labels.Collection is an unnecessary wrapping unless i miss anything. we always do a in place `{}` construction and also only use one from that.

just a thought, not something need to address in the PR. might something worth future clean up

Reply:
I agree, that needs an interface change to `GetProxyWorkloadLabels` - that is why I did not do in this PR but we should clean that up. Only advantage that gives today is it provides utility functions for comparison like `SubsetOf` which can be easily implemented without that wrapper.